### PR TITLE
[KernelGen][Nvidia] Add special_chebyshev_polynomial_v operator

### PR DESCRIPTION
**Pull request by XDYuanzhuLee**
## Summary
- Adds Triton kernel implementation for `torch.special.chebyshev_polynomial_v` (Chebyshev polynomial of the third kind)
- Kernel computes `cos((n + 0.5) * arccos(x)) / cos(arccos(x) / 2)` in float32 for numerical stability
- Tests cover all float dtypes (18 cases); benchmark runs float32 only (torch reference limitation)

## Performance
Test command: `pytest benchmark/test_special_chebyshev_polynomial_v.py --level core` (NVIDIA H20)

| Size | dtype | Torch Latency (ms) | Gems Latency (ms) | Speedup | TFLOPS |
|------|-------|--------------------|--------------------|---------|---------|
| [1073741824] | float32 | 9.451520 | 9.509888 | 0.994 | 0.226 |
| [64, 64] | float32 | 0.009216 | 0.007168 | 1.286 | 0.001 |
| [4096, 4096] | float32 | 0.158720 | 0.165888 | 0.957 | 0.202 |
| [64, 512, 512] | float32 | 0.158720 | 0.164864 | 0.963 | 0.204 |
| [1024, 1024, 1024] | float32 | 9.443840 | 9.593856 | 0.984 | 0.224 |
| **Arithmetic Mean** | — | — | — | **1.037** | — |

## Testing
- 18 accuracy tests all pass
- Validated against PyTorch reference on CUDA device
- Covers float16, float32, bfloat16 dtypes

## Multi-backend Testing

| Backend | Accuracy | Performance |
|---------|----------|-------------|
| NVIDIA H20 | PASS (18/18) | PASS |

## Files Changed
- `src/flag_gems/ops/special_chebyshev_polynomial_v.py` — Triton kernel + wrapper
- `tests/test_special_chebyshev_polynomial_v.py` — accuracy tests
- `benchmark/test_special_chebyshev_polynomial_v.py` — performance benchmark
- `src/flag_gems/ops/__init__.py` — register import
- `src/flag_gems/__init__.py` — register in `_FULL_CONFIG`
- `conf/operators.yaml` — operator metadata

### CODE REVIEW

**Comment by XDYuanzhuLee** [comment]
> ## Description
> Add Triton kernel implementation for `torch.special.chebyshev_polynomial_v` — the Chebyshev polynomial of the third kind V_n(x).
> 
> The kernel computes `cos((n + 0.5) * arccos(x)) / cos(arccos(x) / 2)` in float32 for numerical stability, then casts back to the input dtype.
> 
> ## Changes
> * `src/flag_gems/ops/special_chebyshev_polynomial_v.py` — Triton kernel + wrapper
> * `tests/test_special_chebyshev_polynomial_v.py` — accuracy tests (all float dtypes)
> * `benchmark/test_special_chebyshev_polynomial_v.py` — performance benchmark (float32 only, torch reference limitation)
> * `src/flag_gems/ops/__init__.py` — register import
> * `src/flag_gems/__init__.py` — register in `_FULL_CONFIG`
> * `conf/operators.yaml` — operator metadata
> 
> ## Labels
> KernelGen

PR description is non-compliant

File: conf/operators.yaml [hunk 1]
```diff
@@ -5806,6 +5806,18 @@ ops:
       - DSA
     stages:
       - beta: '5.0'
+  - id: special_chebyshev_polynomial_v
+    description: |
+      Triton kernel implementation for special_chebyshev_polynomial_v.
+    for:
+      - special_chebyshev_polynomial_v
```
Comment (tengqm) [inline conf/operators.yaml:6523]:
```suggestion
      - special.chebyshev_polynomial_v
```

File: src/flag_gems/ops/special_chebyshev_polynomial_v.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import triton
+import triton.language as tl
+
+from flag_gems.utils import pointwise_dynamic, tl_extra_shim
+
+logger = logging.getLogger(__name__)
+
+_cos = tl_extra_shim.cos
+_acos = tl_extra_shim.acos
```
Comment (tengqm) [inline src/flag_gems/ops/special_chebyshev_polynomial_v.py]: Move this into the function body?
I really hate global vars

File: src/flag_gems/__init__.py [hunk 3]
```diff
@@ -488,6 +488,7 @@ def torch_ge(v):
     ("softshrink.out", softshrink_out),
     ("sort", sort),
     ("sort.stable", sort_stable),
+    ("special_chebyshev_polynomial_v", special_chebyshev_polynomial_v),
```
Comment (tengqm) [inline src/flag_gems/__init__.py:545]: Not sure if this should be written in a different way.

```suggestion
    ("special.chebyshev_polynomial_v", special_chebyshev_polynomial_v),
```

File: conf/operators.yaml (hunk 1 above)
Comment (bwbwzzz) [reply conf/operators.yaml:6523]: Fixed. Changed to `special.chebyshev_polynomial_v`.

File: src/flag_gems/ops/special_chebyshev_polynomial_v.py (hunk 2 above)
Comment (bwbwzzz) [reply src/flag_gems/ops/special_chebyshev_polynomial_v.py]: Moved into the kernel function body. Now calls `tl_extra_shim.cos` / `tl_extra_shim.acos` directly inline.

File: src/flag_gems/__init__.py (hunk 3 above)
Comment (bwbwzzz) [reply src/flag_gems/__init__.py:545]: Fixed. Changed to `special.chebyshev_polynomial_v`.

**Comment by bin913** [comment]
@XDYuanzhuLee Please fix failing check

File: src/flag_gems/__init__.py (hunk 3 above)
Comment (XDYuanzhuLee) [reply src/flag_gems/__init__.py:545]: I tried switching this dispatch key to `special.chebyshev_polynomial_v`, but CI `python-op` failed. PyTorch exposes the actual aten operator as `aten::special_chebyshev_polynomial_v`, not `aten::special.chebyshev_polynomial_v` (`torch.ops.aten.special_chebyshev_polynomial_v` exists, while the dotted name does not). With the dotted key, FlagGems does not register the CUDA implementation and the test falls back to PyTorch native CUDA, which fails for Half/BFloat16. I restored the underscore dispatch key in `b43f4dea9` for runtime and CI correctness.

File: conf/operators.yaml (hunk 1 above)
Comment (XDYuanzhuLee) [reply conf/operators.yaml:6523]: I tried switching this dispatch key to `special.chebyshev_polynomial_v`, but CI `python-op` failed. PyTorch exposes the actual aten operator as `aten::special_chebyshev_polynomial_v`, not `aten::special.chebyshev_polynomial_v` (`torch.ops.aten.special_chebyshev_polynomial_v` exists, while the dotted name does not). With the dotted key, FlagGems does not register the CUDA implementation and the test falls back to PyTorch native CUDA, which fails for Half/BFloat16. I restored the underscore dispatch key in `b43f4dea9` for runtime and CI correctness.

**Comment by XDYuanzhuLee** [comment]
Follow-up on the failing check: I tried the dotted dispatch key requested in review, but CI `python-op` failed because PyTorch exposes the aten operator as `aten::special_chebyshev_polynomial_v`, not `aten::special.chebyshev_polynomial_v`. I restored the underscore dispatch key in `b43f4dea9`; the latest `python-op`, `code-style`, `preprocess`, and `triage` checks are passing.

File: src/flag_gems/ops/special_chebyshev_polynomial_v.py (hunk 2 above)
Comment (XDYuanzhuLee) [reply src/flag_gems/ops/special_chebyshev_polynomial_v.py]: Done. Removed the module-level `_cos`/`_acos` aliases; the kernel now calls the shim functions directly inside the function body.

File: conf/operators.yaml (hunk 1 above)
Comment (tengqm) [reply conf/operators.yaml:6523]: THanks for the double confirm.

File: src/flag_gems/__init__.py (hunk 3 above)
Comment (tengqm) [reply src/flag_gems/__init__.py:545]: Thanks for the verification.

File: tests/test_special_chebyshev_polynomial_v.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import torch
+
+import flag_gems
+
+from . import accuracy_utils as utils
+
+
+@pytest.mark.special_chebyshev_polynomial_v
+@pytest.mark.parametrize("shape", utils.POINTWISE_SHAPES)
+@pytest.mark.parametrize("dtype", utils.FLOAT_DTYPES)
```
Comment (tengqm) [inline tests/test_special_chebyshev_polynomial_v.py]: This test won't work for nvidia, right?
In the benchmark, you mentioned that the torch reference implementation only supports fp32.

Comment (XDYuanzhuLee) [reply tests/test_special_chebyshev_polynomial_v.py]: Fixed. Restricted test dtype parametrization to `[torch.float32]` because the torch reference only supports float32 on CUDA, matching the benchmark which already uses `dtypes=[torch.float32]`. Validated with `check_operator.py` (dtype error resolved), `pytest --collect-only` (6 tests), and `pre-commit` (pass). Pushed.

**Review by bin913 — APPROVED** [review]
